src/components/FilterPanel/FilterAccordion.jsx

```diff
@@ -0,0 +1,68 @@
+import {
+  Accordion,
+  AccordionSummary,
+  AccordionDetails,
+  Typography,
+} from '@mui/material';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import { PriceSlider } from './PriceSlider';
+import { CheckboxGroup } from './CheckboxGroup';
+
+export const FilterAccordion = ({
+  section,
+  watchedValues,
+  handlePriceChange,
+  getStateBySection,
+  getHandlerBySection,
+}) => {
+  const renderSectionContent = () => {
+    if (section.type === 'slider') {
+      return (
+        <PriceSlider
+          priceRange={watchedValues.priceRange || [50, 500]}
+          handlePriceChange={handlePriceChange}
+        />
+      );
+    }
+
+    const state = getStateBySection(section.id);
+    const handler = getHandlerBySection(section.id);
+
+    return (
+      <CheckboxGroup
+        section={section}
+        state={state}
+        handler={handler}
+      />
+    );
+  };
```

Comment:
A function that returns a JSX template is a React component. Treat it as such - declare it separately.